addons/survey/models/survey_user_input.py

```diff
@@ -591,10 +633,35 @@ class SurveyUserInputLine(models.Model):
     value_text_box = fields.Text('Free Text answer')
     suggested_answer_id = fields.Many2one('survey.question.answer', string="Suggested answer")
     matrix_row_id = fields.Many2one('survey.question.answer', string="Row answer")
+    answer_display_name = fields.Char('Answer', compute="_compute_answer_display_name",
+        help="Helper field that shows the user's answer as readable text for any type of question")
     # scoring
     answer_score = fields.Float('Score')
     answer_is_correct = fields.Boolean('Correct')
 
+    @api.depends('answer_type')
+    def _compute_answer_display_name(self):
+        for line in self:
+            if line.answer_type == 'char_box':
+                line.answer_display_name = line.value_char_box
+            elif line.answer_type == 'text_box':
+                line.answer_display_name = line.value_text_box[:50] if line.value_text_box else False
+            elif line.answer_type == 'numerical_box':
+                line.answer_display_name = line.value_numerical_box
+            elif line.answer_type == 'date':
+                line.answer_display_name = fields.Date.to_string(line.value_date)
+            elif line.answer_type == 'datetime':
+                line.answer_display_name = fields.Datetime.to_string(line.value_datetime)
+            elif line.answer_type == 'suggestion':
+                if line.matrix_row_id:
+                    line.answer_display_name = '%s: %s' % (
+                        line.suggested_answer_id.value,
+                        line.matrix_row_id.value)
+                else:
+                    line.answer_display_name = line.suggested_answer_id.value
+            else:
+                line.answer_display_name = False
```

Comment:
Would fallback on user_input_id, to match _rec_name default behavior

Reply:
And/Or maybe include the "skipped" mark somehow, as if we have no answer_type this is because we skipped it